src/Classes/Entities/EventEntity.php

```diff
@@ -0,0 +1,226 @@
+
+class EventEntity
+{
+    protected $name;
+    protected $category;
+    protected $location;
+    protected $date;
+    protected $startTime;
+    protected $endTime;
+    protected $notes;
+
+    public function __construct(
+        string $name = null,
+        int $category = null,
+        string $location = null,
+        string $date = null,
+        string $startTime = null,
+        string $endTime = null,
+        string $notes = null
+    ) {
+        $this->name = ($this->name ?? $name);
+        $this->category = ($this->category ?? $category);
+        $this->location = ($this->location ?? $location);
+        $this->date = ($this->date ?? $date);
+        $this->startTime = ($this->startTime ?? $startTime);
+        $this->endTime = ($this->endTime ?? $endTime);
+        $this->notes = ($this->notes ?? $notes);
+
+        $this->sanitiseData();
+    }
+
+    /**
+     * Will sanitise all the fields for an event.
+     */
+    private function sanitiseData()
+    {
+        $this->name = $this->sanitiseString($this->name);
+        $this->name = self::validateExistsAndLength($this->name, 255);
+        $this->category = (int)$this->category;
+        $this->location = $this->sanitiseString($this->location);
+        $this->location = self::validateExistsAndLength($this->location, 255);
+        $this->date = $this->validateDate($this->date);
+        $this->startTime = $this->validateTime($this->startTime);
+        $this->endTime = $this->validateTime($this->endTime);
+        $this->validateStartEndTime($this->startTIme, $this->endTime);
+        $this->notes = $this->sanitiseString($this->notes);
+        $this->notes = self::validateLength($this->notes, 5000);
+    }
+
+    /**
+     * Sanitise as a date in the event table as data.
+     *
+     * @param string $date
+     *
+     * @return bool|string
+     */
+    public static function validateDate(string $date)
+    {
+        if (!preg_match('/([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/', $date)) {
+            throw new \Exception('Please enter correct date');
+        } else {
+            return $date;
+        }
+    }
+
+    /**
+     * Sanitise as a time in the event table as data.
+     *
+     * @param string $time
+     * @return bool|string
+     */
+    public static function validateTime(string $time)
+    {
+        if (!preg_match('/([01][0-9]|2[0-3]):[0-5][0-9]/', $time)
+        ) {
+            throw new \Exception('Please enter correct time');
+        } else {
+            return $time;
+        }
+    }
+
+    /**
+     * Validate that end time is later than start time
+     *
+     * @param $startTime
+     * @param $endTime
+     * @return bool
+     * @throws \Exception
+     */
+    public static function validateStartEndTime($startTime, $endTime)
+    {
+        if ($startTime >= $endTime) {
+            throw new \Exception('End time should be later than start time');
+        } else {
+            return true;
+        }
+    }
```

Comment:
How does this actually validate time?

Reply:
I have not completely understood why but my testing says that it is just doing what is required for our uses

Reply:
A very good question @mallowmew . It will just be comparing the values as strings typecast to integers which fortuitously works ok for times in this format (I think!).
If you were being thorough you would convert to dates and use date_diff or summat.